sdk/core/azure-core-http-jdk11-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/JdkAsyncHttpClient.java

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.jdk.httpclient;
+
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.HttpHeader;
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.FluxUtil;
+import reactor.adapter.JdkFlowAdapter;
+import reactor.core.Exceptions;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.net.URISyntaxException;
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.util.Collections;
+import java.util.List;
+
+import java.net.http.HttpRequest.BodyPublisher;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.Flow;
+
+import static java.net.http.HttpResponse.BodyHandlers.*;
+import static java.net.http.HttpRequest.BodyPublishers.*;
+
+/**
+ * HttpClient implementation for the JDK 11 HttpClient.
+ */
+class JdkAsyncHttpClient implements HttpClient {
+    private final java.net.http.HttpClient jdk11HttpClient;
+
+    private static final Set<String> JDK11_RESTRICTED_HEADERS;
+    static {
+        TreeSet<String> treeSet = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
+        treeSet.addAll(Set.of("connection",
+            "content-length",
+            "date", // BUG: Java11: https://bugs.openjdk.java.net/browse/JDK-8213189 (Fixed in JDK12)
```

Comment:
How will this work for libraries which require fine tuning for these headers? In the case of Storage the upload could fail on a mismatch `Content-Length` header and shared key credential authentication will fail if the `Date` header is mismatch compared to what was used to generate the key.

Reply:
If JDK httpClient supports completely disabling its header checks programmatically then that would be the easiest path. So far we didn't see that Netty/OkHttp blocking any headers.

If full disabling is not possible then we need to see if the native HTTP client has a programmatic way to white list subset of headers, if yes we can identify the headers that JDK blocks by default but required by our SDKs to function properly and white list them.

Yet to explore.

Reply:
Relaxed validations and added necessary log messages as agreed offline. 

Regarding the headers we care about:

`Date` header is allowed from JDK12+
`Content-Length` header will be set by JDK native HttpClient before sending out the requests.